### DISCUSSION TITLE
Invalid dependency (must be a url...

### DISCUSSION THREAD

**Discussion by nathanjbeaumont** [JuliaHealth/KomaMRI.jl#504] [Q&A] [answered]
Hello all,
I did a fresh install of Julia, then did `add KomaMRI` and `add CUDA` in the package manager. Then did:
```
julia> using KomaMRI, CUDA

julia> KomaUI()
```
and this error occurs in the terminal:
`ERROR: Invalid dependency (must be a url, file, or AssetRegistry path): C:\Users\natha\.julia\packages\Interact\PENUy\src\..\assets\bulma.min.css`

When I interact with many things in the GUI the same error occurs. I clicked "Simulate!" and the spinning wheel shows indefinitely, although the console progress bar works and goes to 100%. I clicked Phantom -> view phantom and "plotting phantom" shows on screen indefinitely with a spinning wheel. I clicked "Reconstruct!" and the wheel shows indefinitely but reconstruction never seems to complete. Unsure what's related to the error.

**Comment by cncastillo** [comment]
Hi! Thanks for reporting this. This might be a problem with Julia 1.11. 

What Julia version are you using? And what version of Koma?

I will see if I can reproduce the problem and potentially find a solution.

**Reply by nathanjbeaumont** [reply]
Julia Version 1.11.1 (2024-10-16), KomaMRI v0.9.0 [6a340f8b]

**Comment by cncastillo** [comment]
I used Julia 1.11.1 and KomaMRI v0.9.0 and could not reproduce this problem on Linux. 

Steps:
- Created new folder (`mkdir KomaUIBug`)
- Activated folder environment and opened Julia (`julia +release --project=KomaUIBug`)
- `using KomaMRI, CUDA` and then `KomaUI` and pressed simulate and reconstruct.

```julia
(KomaUIBug) pkg> st
Status `~/Documents/KomaUIBug/Project.toml`
  [052768ef] CUDA v5.5.2
  [6a340f8b] KomaMRI v0.9.0
```

Could you do that and attach the Manifest.toml here?

**Reply by nathanjbeaumont** [reply]
[Manifest.txt](https://github.com/user-attachments/files/17593936/Manifest.txt)

See attached. I had to change the .toml to .txt to upload here

**Comment by cncastillo** [comment]
BTW in the meantime, you could try using Julia 1.10 and try again.

In the terminal:

```bash
juliaup add 1.10
juliaup default 1.10
julia
```

As Julia 1.11 has been a little unstable, a new version might be coming soon.

**Reply by nathanjbeaumont** [reply]
I uninstalled Julia 1.11.1 via the windows uninstaller, then I installed Julia 1.10.6 (LTS) via the 64 bit executable here: https://julialang.org/downloads/

Opened the Julia terminal, pressed `]` to enter the package manager, did `add CUDA` then `add KomaMRI`, cntrl+c to exit the package manager, then did `using KomaMRI, CUDA`, `KomaUI()`. Getting the exact same error:
`ERROR: Invalid dependency (must be a url, file, or AssetRegistry path): C:\Users\natha\.julia\packages\Interact\PENUy\src\..\assets\bulma.min.css`

Also getting these warnings after doing `add KomaMRI`
![image](https://github.com/user-attachments/assets/7f2365ea-c5da-4768-aed9-3015a8e8f2af)

project status is:
```
Status `C:\Users\natha\.julia\environments\v1.10\Project.toml`
  [052768ef] CUDA v5.5.2
  [6a340f8b] KomaMRI v0.9.0
```

**Reply by nathanjbeaumont** [reply]
Manifest.toml for the v1.10 project is here:
[Manifest.txt](https://github.com/user-attachments/files/17594019/Manifest.txt)

**Reply by cncastillo** [reply]
That's very weird. Can you install Julia through the Windows store? That is the recommended way. I have seen some people having problems installing it with the executable.

**Reply by nathanjbeaumont** [reply]
When I had julia 1.11.1 initially I had installed it through the windows store with `winget install julia -s msstore`

**Reply by cncastillo** [reply, marked as answer]
In this issue https://github.com/JuliaGizmos/Interact.jl/issues/406 they recommend to do `] build`, but the user with the problem after says:
> "Apparently it was the: C:\Users\Stephen.julia\registries\General - folder which was the problem, which I have deleted!". 

So, something might be corrupted. You can try the `] build` command as well.

As I can't reproduce this problem I think I would advice you to: uninstalling Julia using the Windows store, delete the whole `.julia` folder, restart PC (Windows sometimes requires this), install Julia through the Windows store (the app, not cmd), restart (not sure if this one is required), install KomaMRI again.

Sorry for the bad experience. If that doesn't work I will ask in the Julia Slack to see if they can help us, but this not necessarily a bug with Koma, it could be a problem in Interact.jl, or as I said something might be slightly corrupted in the Julia installation. 

You can also do simulations without the UI, using KomaMRICore, KomaMRIPlots, KomaMRIFiles. Some examples here: https://juliahealth.org/KomaMRI.jl/stable/tutorial/03-ChemicalShiftEPI/. Just replace `using KomaMRI` by `using KomaMRICore, KomaMRIPlots, KomaMRIFiles`. These do not use Interact.jl, so that error will not occur.

**Reply by nathanjbeaumont** [reply]
Good news, uninstalling Julia, then deleting `C:\Users\natha\.julia`, then installing Julia through the Microsoft Store fixed the issue.

I think the reason reinstalling previously didn't work is because uninstalling via "Windows Settings > Apps > Installed Apps" doesn't delete the .julia folder. Thanks for the help.

**Reply by cncastillo** [reply]
Amazing :)! I'm glad it worked. Let me know if you have any other problem!

**Reply by nathanjbeaumont** [reply]
Having this issue with spiral.seq now: https://github.com/JuliaHealth/KomaMRI.jl/discussions/505
